server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1917,19 +1920,22 @@ public boolean isUpgrading() {
     return upgradeCoordinator.getStatus() != UpgradeCoordinator.UpgradeStatus.COMPLETE;
   }
 
-  void initializeBalancer() {
+  private void initializeBalancer() {
+    String configuredBalancerClass = getConfiguration().get(Property.MANAGER_TABLET_BALANCER);
     try {
-      getContext().getPropStore().getCache().removeAll();
-      getConfiguration().invalidateCache();
-      log.debug("Attempting to reinitialize balancer using class {}",
-          getConfiguration().get(Property.MANAGER_TABLET_BALANCER));
-      var localTabletBalancer = Property.createInstanceFromPropertyName(getConfiguration(),
-          Property.MANAGER_TABLET_BALANCER, TabletBalancer.class, new DoNothingBalancer());
-      localTabletBalancer.init(balancerEnvironment);
-      tabletBalancer = localTabletBalancer;
+      if (tabletBalancer == null
+          || !tabletBalancer.getClass().getName().equals(configuredBalancerClass)) {
+        log.debug("Attempting to initialize balancer using class {}, was {}",
+            configuredBalancerClass,
+            tabletBalancer == null ? "null" : tabletBalancer.getClass().getName());
+        var localTabletBalancer = Property.createInstanceFromPropertyName(getConfiguration(),
+            Property.MANAGER_TABLET_BALANCER, TabletBalancer.class, new DoNothingBalancer());
+        localTabletBalancer.init(balancerEnvironment);
+        tabletBalancer = localTabletBalancer;
```

Comment:
There used to be the following log message from ManagerClientService.updatePlugins that logged something at info once the change was complete.  Would be nice to still have that.  The following probably will not compile, copied the log message exactly from the old place in the code.

```suggestion
        tabletBalancer = localTabletBalancer;
        log.info("tablet balancer changed to {}", manager.getBalancerClass().getName());
```